server/service/src/requisition/request_requisition/insert_program.rs

```diff
@@ -63,26 +66,66 @@ pub fn insert_program_request_requisition(
 }
 
 fn validate(
-    connection: &StorageConnection,
-    store_id: &str,
+    ctx: &ServiceContext,
     input: &InsertProgramRequestRequisition,
-) -> Result<(), OutError> {
+) -> Result<(ProgramRow, ProgramRequisitionOrderTypeRow), OutError> {
+    let connection = &ctx.connection;
+
     if let Some(_) = check_requisition_exists(connection, &input.id)? {
         return Err(OutError::RequisitionAlreadyExists);
     }
 
-    // TODO: Check if we already have reached out max requisitions for this period
-    // https://github.com/openmsupply/open-msupply/issues/1599
+    let requisitions = RequisitionRepository::new(&connection).query_by_filter(
+        RequisitionFilter::new().order_type(EqualFilter::equal_to(&input.program_order_type_id)),
+    )?;
 
-    // TODO: Check Order Type is valid for this Program
-    // https://github.com/openmsupply/open-msupply/issues/1599
+    let program_settings = get_program_requisition_settings(ctx, &ctx.store_id)?;
 
-    // TODO: Check if the 'other_party' is a valid Program Supplier
-    // https://github.com/openmsupply/open-msupply/issues/1599
+    let order_type: ProgramRequisitionOrderTypeRow = program_settings
+        .iter()
+        .map(|setting| {
+            setting
+                .order_types
+                .iter()
+                .filter(|order_type| order_type.order_type.id == input.program_order_type_id)
+                .map(|order_type| order_type.order_type.clone())
+        })
+        .flatten()
+        .next()
+        .ok_or(OutError::ProgramOrderTypeDoesNotExist)?;
+
+    if requisitions.len() as i32 >= order_type.max_order_per_period {
+        return Err(OutError::MaxOrdersReachedForPeriod(requisitions));
+    }
+
+    let program_requisition_settings: ProgramRequisitionSettingsRow = program_settings
```

Comment:
I think at the start of validation we would just need to find order_type and program from program_settings, validate can return program_setting (to get program_id), order_type (from program_settings, to get the name).

Something like this:

```rust
    let (program_setting, order_type) = program_settings
        .iter()
        .find_map(|setting| {
            setting
                .order_types
                .iter()
                .find(|order_type| order_type.order_type.id == input.program_order_type_id)
                .map(|order_type| (setting.clone(), order_type.clone()))
        })
        .ok_or(OutError::ProgramOrderTypeDoesNotExist)?;


// check that period_id is in order_type.available_periods.iter()..

// check that supplier in program_setting
```